### PR TITLE
feat(shared): Introduce isDevelopmentEnvironment() helper to check identify application runtime mode

### PR DESCRIPTION
**Pull request by dimkl**
## Description

Identify development mode using the `process.env` or `import.meta.env`.

## Checklist

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.
- [x] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [x] (If applicable) [Documentation](https://github.com/clerkinc/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [x] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

### CODE REVIEW

**Review by anagstef — APPROVED** [review]
🔥 🔥 🔥

File: packages/shared/src/utils/runtimeMode.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+export const isDevelopmentMode = (): boolean => {
```
Comment (LekoArts) [inline packages/shared/src/utils/runtimeMode.ts:1]: Nit: Node.js calls this "development and production environments" (https://nodejs.dev/en/learn/nodejs-the-difference-between-development-and-production/). So a suggestion for another name would be `isDevelopmentEnvironment`

Comment (dimkl) [reply packages/shared/src/utils/runtimeMode.ts]: helper renamed. ptal

**Comment by clerk-cookie** [comment]
This PR has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.

**Pull request merged**